com.stansassets.build/Editor/Steps/UnityPlayerBuildStep.cs

```diff
@@ -49,59 +49,76 @@ private void BuildProject()
         }
 
-        private void SetContext()
+        void SetContext()
         {
             for (int i = 0; i < s_Tasks.Count; i++)
             {
                 s_Tasks[i].SetContext(m_BuildContext);
             }
         }
-        
+
         [PostProcessScene(1)]
-        private static void RunTasks()
+        static void RunTasks()
         {
             if(Application.isPlaying)
                 return;
-            
+
             Scene currentScene = SceneManager.GetActiveScene();
             GameObject[] rootGameObjects = currentScene.GetRootGameObjects();
-
-            List<GameObject> currentSceneObjects = GetAllObjects(rootGameObjects);
+            Dictionary<GameObject, List<Component>> componentsMap = FetchComponentsMap(rootGameObjects);
 
             for (int i = 0; i < s_Tasks.Count; i++)
             {
                 s_Tasks[i].OnPostprocessScene(currentScene);
-                for (int j = 0; j < currentSceneObjects.Count; j++)
+                foreach (var pair in componentsMap)
                 {
-                    List<Component> allObjectComponents = GetAllComponents(currentSceneObjects[j]);
-                    s_Tasks[i].OnPostprocessGameObject(currentSceneObjects[j], allObjectComponents);
+                    var gameObject = pair.Key;
+                    var components = pair.Value;
+
+                    if (gameObject == null)
+                        continue;
+
+                    s_Tasks[i].OnPostprocessGameObject(gameObject, components);
                 }
             }
+
         }
 
-        private static List<GameObject> GetAllObjects(GameObject[] rootGameObjects)
+        static Dictionary<GameObject, List<Component>> FetchComponentsMap(GameObject[] rootGameObjects)
         {
-            List<GameObject> allObjects = new List<GameObject>();
-            
+            var componentsMap = new Dictionary<GameObject, List<Component>>();
             for (int i = 0; i < rootGameObjects.Length; i++)
             {
-                var componentsInChildren =
-                    rootGameObjects[i].GetComponentsInChildren<Transform>();
+                if(rootGameObjects[i] == null)
+                    continue;
 
-                for (int j = 0; j < componentsInChildren.Length; j++)
-                {
-                    allObjects.Add(componentsInChildren[j].gameObject);
-                }
+                StoreObjectsAndComponents(rootGameObjects[i], componentsMap);
             }
-
-            return allObjects;
+            return componentsMap;
         }
-        
-        private static List<Component> GetAllComponents(GameObject rootObject)
+
+        static readonly List<Transform> s_TempTransformsCollection = new List<Transform>();
+        static readonly List<Component> s_TempComponentsCollection = new List<Component>();
+        static void StoreObjectsAndComponents(GameObject rootGameObject, Dictionary<GameObject, List<Component>> componentsMap)
```

Comment:
Please, consider changing the name of this method to something like `FetchHierarchy ` to be more informative